### DISCUSSION TITLE
What is the logic for a private code repository to be indexed?

### DISCUSSION THREAD

**Discussion by ali50m** [community/community#57889] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I started a private dotnet code repo in github about two years ago, which named like **MyProject**. It was indexed and I can do the code research easily inside github.

About two and half months ago, I decided to rename the repo to MyProject-orig, and reupload the same repo with clean fresh git history (removed the .git folder and recreated the repo), and still with the name of **MyProject**.

My issue is, after 2.5 months later, the new created **MyProject** repo is still unindexed. I worked actively on this repo and made about 300 commits to it. It is inconvenient that can not do the code search on this repo.

Is there something I missed for indexing this repo? Please help me out, thanks!

**Comment by Gustavonn07** [comment, marked as answer]
It is possible that the new repository has not yet been indexed by the GitHub search engine. Normally, GitHub indexes repositories automatically, but there may be cases where this does not happen immediately or there may be a delay in the indexing process.

Here are some suggestions to help resolve the issue:

1. Check the repository visibility settings: Make sure the new repository is set to private so that indexing takes place correctly.

2. Wait longer: Sometimes it can take some time for GitHub to fully index a repository, especially if there are a large number of commits or recent changes. Wait a few more days to see if the repository is indexed.

if the problem persists it may be better to contact github support: https://support.github.com/

**Reply by ali50m** [reply]
Thanks!

**Comment by thibauds** [comment]
This happened to me with all the repos that were transferred. Doing a search in those repos started the indexing process.
